### PR TITLE
Nb as data api

### PR DESCRIPTION
**Pull request by ewmassey**
This PR contains the creation of the Notebook and NotebookCollection classes that are used for analyzing notebooks. Other notable changes:

- Notebook and NotebookCollection classes introduced
- configuration handled by summary object in papermill.conf
- broke out reading and writing into separate module iorw.py
- record_value -> record
- introduced pm.display for displaying outputs with keys so they can be easily recalled later on

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+Papermill is a tool for parameterizing, executing, and analyzing Jupyter Notebooks.
 
-Stepping away from the hyperbole, our goals for Papermill include simplifying
-and streamlining:
+The goals for Papermill are:
 
 * Parametrizing notebooks
 * Executing and collecting metrics across the notebooks
-* Summarize collections of notebooks
+* Summarizing collections of notebooks
```
Comment (rgbkrk) [inline README.rst]: Much improved in this overall section.

File: README.rst [hunk 2]
```diff
@@ -21,51 +20,91 @@ Installation
 Usage
 -----
 
-Executing a parametrized notebook
+Parameterizing a notebook.
```
Comment (rgbkrk) [inline README.rst]: 👍

File: papermill/api.py [hunk 3]
```diff
@@ -1,239 +1,191 @@
...
+        self._notebooks[key] = value
+
+    def __getitem__(self, key):
+        return self._notebooks[key]
+
+    def __delitem__(self, key):
+        del self._notebooks[key]
+
+    @classmethod
+    def from_directory(cls, path):
```
Comment (rgbkrk) [inline papermill/api.py]: I feel like these class methods for initializing should still have a verb in the name. What's the precedent for this?

**Comment by rgbkrk** [comment]
Rockin'

**Comment by adparker** [comment]
Cool!

**Pull request merged**